client/common/svg.download.js

```diff
@@ -0,0 +1,84 @@
+import { select } from 'd3-selection'
+import { to_svg } from '../src/client'
+
+export function downloadChart(self, div, mainG, svgStyle) {
+	if (!self.state) return
+	// has to be able to handle multichart view
+	const mainGs = []
+	const translate = { x: undefined, y: undefined }
+	const titles = []
+	let maxw = 0,
+		maxh = 0,
+		tboxh = 0
+	let prevY = 0,
+		numChartsPerRow = 0
+
+	self.dom[div].selectAll(mainG).each(function() {
+		mainGs.push(this)
+		const bbox = this.getBBox()
+		if (bbox.width > maxw) maxw = bbox.width
+		if (bbox.height > maxh) maxh = bbox.height
+		const divY = Math.round(this.parentNode.parentNode.getBoundingClientRect().y)
+		if (!numChartsPerRow) {
+			prevY = divY
+			numChartsPerRow++
+		} else if (Math.abs(divY - prevY) < 5) {
+			numChartsPerRow++
+		}
+		const xy = select(this)
+			.attr('transform')
+			.split('translate(')[1]
+			.split(')')[0]
+			.split(',')
+			.map(d => +d.trim())
+
+		if (translate.x === undefined || xy[0] > translate.x) translate.x = +xy[0]
+		if (translate.y === undefined || xy[1] > translate.y) translate.y = +xy[1]
+
+		const title = this.parentNode.parentNode.firstChild
+		const tbox = title.getBoundingClientRect()
+		console.log('what is tbox', tbox)
+		if (tbox.width > maxw) maxw = tbox.width
+		if (tbox.height > tboxh) tboxh = tbox.height
+		titles.push({ text: title.innerText, styles: window.getComputedStyle(title), tbox })
+	})
+
+	// add padding between charts
+	maxw += 80
+	maxh += 80
+
+	const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+
+	const svgSel = select(svg)
+		.style('display', 'block')
+		.style('opacity', 1)
+		.attr('width', numChartsPerRow * maxw)
+		.attr('height', Math.floor(mainGs.length / numChartsPerRow) * maxh)
+
+	const svgStyles = window.getComputedStyle(document.querySelector(svgStyle))
+	for (const prop of svgStyles) {
+		if (prop.startsWith('font')) svgSel.style(prop, svgStyles.getPropertyValue(prop))
+	}
+
+	mainGs.forEach((g, i) => {
+		const mainG = g.cloneNode(true)
+		const colNum = i % numChartsPerRow
+		const rowNum = Math.floor(i / numChartsPerRow)
+		const corner = { x: colNum * maxw + translate.x + 80, y: rowNum * maxh + translate.y }
+		console.log('what is titles', titles)
+		const title = select(svg)
+			.append('text')
+			.attr('transform', 'translate(' + (corner.x + titles[i].tbox.width / 2 - 100) + ',' + corner.y + ')')
+			.text(titles[i].text)
+		for (const prop of titles[i].styles) {
+			if (prop.startsWith('font')) title.style(prop, titles[i].styles.getPropertyValue(prop))
+		}
+
+		select(mainG).attr('transform', 'translate(' + corner.x + ',' + (corner.y + tboxh) + ')')
+		svg.appendChild(mainG)
+	})
+
+	//const svg_name = self.config.term.term.name + ' barchart'
+	const svg_name = ' barchart'
```

Comment:
The (optional) filename should be an argument, or default to self.chartType or a non-chart-specific generic name
```js
export function downloadChart(self, div, mainG, svgStyle, filename='') {
    ...
   const svg_name = filename || self.chartType || 'download'
   ...
}
```
It's confusing to see "barchart" in the filename for a downloaded survival plot.

Reply:
Sorry, I just changed it